eventhubs/client/azure-eventhubs/src/main/java/com/azure/eventhubs/ConnectionStringBuilder.java

```diff
@@ -56,15 +57,24 @@
     private static final String SHARED_ACCESS_KEY_CONFIG_NAME = "SharedAccessKey";
     private static final String SHARED_ACCESS_SIGNATURE_CONFIG_NAME = "SharedAccessSignature";
 
-    private static final String ALL_KEY_ENUMERATE_REGEX = "(" + HOST_NAME_CONFIG_NAME + "|" + ENDPOINT_CONFIG_NAME + "|"
-        + SHARED_ACCESS_KEY_NAME_CONFIG_NAME + "|" + SHARED_ACCESS_KEY_CONFIG_NAME + "|"
-        + SHARED_ACCESS_SIGNATURE_CONFIG_NAME + "|" + ENTITY_PATH_CONFIG_NAME + ")";
-
-    private static final String KEYS_WITH_DELIMITERS_REGEX = KEY_VALUE_PAIR_DELIMITER + ALL_KEY_ENUMERATE_REGEX
-            + KEY_VALUE_SEPARATOR;
+    private static final String ALL_KEY_ENUMERATE_REGEX = new StringBuilder()
+        .append("(")
+        .append(HOST_NAME_CONFIG_NAME).append(VERTICAL_LINE)
```

Comment:
I would use `String.join("|")` rather than creating an object and disposing of it.